.github/workflows/tests.yaml

```diff
@@ -175,4 +175,20 @@ jobs:
       - name: >-
           Test outcome: ${{ steps.test.outcome }}
         # every step must define a `uses` or `run` key
-        run: echo NOOP
+        run: echo 
+
+  valgrind:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.0'
+      - run: sudo apt-get install -y valgrind
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('Gemfile') }}
+          restore-keys: ${{ runner.os }}-gems-
+      - run: bundle install --jobs=3 --retry=3 --path=vendor/bundle
```

Comment:
You let setup-ruby both run `bundle install` and cache: https://github.com/ruby/setup-ruby#caching-bundle-install-automatically

Reply:
I made the changes you suggested.